contrib/cirrus/runner.sh

```diff
@@ -278,16 +279,8 @@ logformatter() {
         |& "${CIRRUS_WORKING_DIR}/${SCRIPT_BASE}/logformatter" "$output_name"
 }
 
-# Handle local|remote integration|system testing in a uniform way
-dotest() {
-    local testsuite="$1"
-    req_env_vars testsuite CONTAINER TEST_ENVIRON PRIV_NAME
-
-    # shellcheck disable=SC2154
-    if ((CONTAINER==0)) && [[ "$TEST_ENVIRON" == "container" ]]; then
-        exec_container  # does not return
-    fi;
-
+# reexec the runner.sh scrip tas rootless user if necessary
+reexec_if_rootless() {
```

Comment:
My proposed diff. will make rootless-mode available to all tests, but your way will work as well.  My thinking was: If two runner functions now need to run rootless, chances are good a third (and fourth and fifth and...) may be coming in the future.